main.js

```diff
@@ -0,0 +1,313 @@
+   *   input byte size.
+   * @param {boolean} [multibase=false] - use multibase encoding.
+   *
+   * @return {IdEncoder}
+   */
+  constructor({
+    encoding = 'base58',
+    fixedLength = false,
+    fixedBitLength,
+    multibase = false
+  } = {}) {
+    switch(encoding) {
+      case 'hex':
+      case 'base16':
+        this.encoder = _base16Encoder;
+        this.multibasePrefix = 'f';
+        break;
+      case 'base16upper':
+        this.encoder = _base16Encoder;
+        this.multibasePrefix = 'F';
+        break;
+      case 'base58':
+      case 'base58btc':
+        this.encoder = _base58Encoder;
+        this.multibasePrefix = 'z';
+        break;
+      default:
+        throw new Error(`Unknown encoding type: "${encoding}"`);
+    }
+    this.fixedLength = fixedLength || fixedBitLength !== undefined;
+    if(this.fixedLength) {
+      this.fixedBitLength = _calcOptionsBitLength({
+        // default of 0 calculates from input size
+        defaultLength: 0,
+        bitLength: fixedBitLength
+      });
+    }
+    this.encoding = encoding;
+    this.multibase = multibase;
+  }
+
+  /**
+   * Encode id bytes into a string.
+   *
+   * @param {Uint8Array} bytes - bytes to encode.
+   *
+   * @return {string}
+   */
+  encode(bytes) {
+    const encoded = this.encoder({bytes, idEncoder: this});
+    if(this.multibase) {
+      return this.multibasePrefix + encoded;
+    }
+    return encoded;
+  }
+}
+
+export class IdDecoder {
+  /**
+   * Creates a new IdDecoder instance.
+   *
+   * An IdDecoder decodes an id string into a byte array. It is recommended to
+   * use the fixedBitLength option to avoid padding ids resulting in a larger
+   * than expected byte length.
+   *
+   * @param {string} [encoding='base58'] - encoding format. Ignored if
+   *   multibase is true.
+   * @param {Number} [fixedBitLength] - fixed output bit length. Values with
+   *   leading non-zero data will error.
+   * @param {boolean} [multibase=false] - use multibase encoding to detect
+   *   the id format.
+   *
+   * @return {IdDecoder}
+   */
+  constructor({
+    encoding = 'base58',
+    fixedBitLength,
+    multibase = false
+  } = {}) {
+    this.encoding = encoding;
+    this.fixedBitLength = fixedBitLength;
+    this.multibase = multibase;
+  }
+
+  /**
+   * Decode id string into bytes.
+   *
+   * @param {string} id - id to decode.
+   *
+   * @return {Uint8Array}
+   */
+  decode(id) {
+    let encoding;
+    let data;
+    if(this.multibase) {
+      if(id.length < 1) {
+        throw new Error('Multibase encoding not found.');
+      }
+      const prefix = id[0];
+      data = id.substring(1);
+      switch(id[0]) {
+        case 'f':
+          encoding = 'base16';
+          break;
+        case 'F':
+          encoding = 'base16upper';
+          break;
+        case 'z':
+          encoding = 'base58';
+          break;
+        default:
+          throw new Error(`Unknown multibase prefix "${prefix}"`);
```

Comment:
period.  Assuming a period is added here, there are other places punctuation is needed.